recipes/_omnibus_toolchain.rb

```diff
@@ -21,6 +21,11 @@
 toolchain_name = node['omnibus']['toolchain_name']
 toolchain_version = node['omnibus']['toolchain_version']
 
+if File.exist?('/opt/omnibus-toolchain') || File.exist?('/opt/angry-omnibus-toolchain')
```

Comment:
Code flow wise we should drop this `installed` variable and just place everything after this in an `else`:

```ruby
if File.exist?('/opt/omnibus-toolchain') || File.exist?('/opt/angry-omnibus-toolchain')
  Chef::Log.warn('Assuming the existence of /opt/(angry-)omnibus-toolchain means that the package is already installed.')
else
    # ...
    remote_file package_path do
    source package_url
    action :create_if_missing
    not_if { installed }
  end
  
  package node['omnibus']['toolchain_name'] do
    source package_path
    options install_options
    not_if { installed }
  end
end
```

All that being said `remote_file` and `package` should be idempotent so placing _any_ extra guard logic in seems frivolous.